parsers/US_SPP.py

```diff
@@ -0,0 +1,198 @@
+#!usr/bin/env python3
+
+"""Parser for the Southwest Power Pool area of the United States."""
+
+from dateutil import parser, tz
+from io import StringIO
+from logging import getLogger
+import pandas as pd
+import requests
+
+generation_url = 'https://marketplace.spp.org/public-data-api/gen-mix/asFile'
+
+exchange_url = 'https://marketplace.spp.org/public-data-api/interchange-trend/asFile'
+
+mapping = {'coal': 'coal',
+           'unknown': 'unknown',
+           'Wind': 'wind',
+           'Nuclear': 'nuclear',
+           'Hydro': 'hydro',
+           'Solar': 'solar',
+           'Natural Gas': 'gas',
+           'Diesel Fuel Oil': 'oil',
+           'Waste Disposal Services': 'biomass'
+            }
+
+tie_mapping = {'US-MISO->US-SPP': ['AMRN', 'DPC', 'GRE', 'MDU', 'MEC', 'NSP', 'OTP']}
+
+# NOTE
+# Data sources return timestamps in GMT.
+# Energy storage situation unclear as of 16/03/2018, likely to change quickly in future.
+
+
+def get_data(url, session=None):
+    """Returns a pandas dataframe."""
+
+    s=session or requests.Session()
+    req = s.get(url, verify=False)
+    df = pd.read_csv(StringIO(req.text))
+
+    return df
+
+
+def data_processor(df, logger):
+    """
+    Takes a dataframe and logging instance as input.
+    Checks for new generation types and logs awarning if any are found.
+    Parses the dataframe row by row removing unneeded keys.
+    Returns a list of 2 element tuples, each containing a datetime object
+    and production dictionary.
+    """
+
+    # Remove leading whitespace in column headers.
+    df.columns = df.columns.str.strip()
+    print(df)
```

Comment:
Maybe remove this?

Reply:
:astonished: whoops that was left over from debugging.